### PR TITLE
Addd center_of_charge attribute

### PR DESCRIPTION
**Pull request by PicoCentauri**
Added a `center_of_charge` attribute to groups. Also added the equations for `center_of_mass` and `center_of_geometry` to the documentation.

Is there an additional spot in the docs to mention that a `center_of_charge` exists?

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?

### CODE REVIEW

File: testsuite/MDAnalysisTests/core/test_topologyattrs.py [hunk 1]
```diff
@@ -35,9 +35,10 @@
 
 import MDAnalysis as mda
 import MDAnalysis.core.topologyattrs as tpattrs
-from MDAnalysis.core import groups
+from MDAnalysis.core._get_readers import get_reader_for
 from MDAnalysis.core.topology import Topology
 from MDAnalysis.exceptions import NoDataError
+from soupsieve import escape
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/core/test_topologyattrs.py]: Why is this here? Can't seem to see where this is being used.

Comment (PicoCentauri) [reply testsuite/MDAnalysisTests/core/test_topologyattrs.py]: Ahh damn. sorry

File: package/MDAnalysis/core/topologyattrs.py [hunk 2]
```diff
@@ -1922,6 +1927,73 @@ def get_segments(self, sg):
...
+            output will be a 2d coordinate array of shape ``(n, 3)`` where
+            ``n`` is the number of compounds.
+
+        Note
+        ----
+        This method can only be accessed if the underlying topology has
+        information about atomic charges.
+        """
+        atoms = group.atoms
+        return atoms.center(weights=atoms.charges.__abs__(),
```
Comment (richardjgowers) [inline package/MDAnalysis/core/topologyattrs.py:1991]: Isn't calling abs this way a little cheeky?

**Review by richardjgowers — APPROVED** [review]
Lgtm, a little question about the absolute call

File: package/MDAnalysis/core/topologyattrs.py (hunk 2 above)
Comment (IAlibay) [reply package/MDAnalysis/core/topologyattrs.py:1991]: Doesn't abs just call `cls.__abs__(self)`? You're cutting out the middleman 😜

Comment (PicoCentauri) [reply package/MDAnalysis/core/topologyattrs.py:1991]: We could also do `np.abs(atoms.charges)` but the method is faster...

```
a = np.array([-1,1,-1,1,-1,1])
%timeit np.abs(a)
2.73 µs ± 981 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

```
%timeit a.__abs__()
1.44 µs ± 128 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

File: package/MDAnalysis/core/topologyattrs.py [hunk 3]
```diff
@@ -1922,6 +1927,73 @@ def get_segments(self, sg):
...
+            If `compound` was set to ``'group'``, the output will be a single
+            position vector.
+            If `compound` was set to ``'segments'`` or ``'residues'``, the
+            output will be a 2d coordinate array of shape ``(n, 3)`` where
+            ``n`` is the number of compounds.
+
+        Note
+        ----
+        This method can only be accessed if the underlying topology has
+        information about atomic charges.
```
Comment (IAlibay) [inline package/MDAnalysis/core/topologyattrs.py:1986]: Versionadded would be good at the end here.

**Review by IAlibay — APPROVED** [review]
Just the one thing from me please.

File: package/MDAnalysis/core/topologyattrs.py (hunk 3 above)
Comment (PicoCentauri) [reply package/MDAnalysis/core/topologyattrs.py:1986]: For sure!

File: package/MDAnalysis/core/topologyattrs.py (hunk 2 above)
Comment (IAlibay) [reply package/MDAnalysis/core/topologyattrs.py:1991]: @richardjgowers you fine with this?

Comment (richardjgowers) [reply package/MDAnalysis/core/topologyattrs.py:1991]: Yeah it's fine

**Comment by orbeckst** [comment]
@richardjgowers @IAlibay I leave it to you to merge as I don't know if there's a particular reason to not merge it yet.

**Comment by IAlibay** [comment]
> @richardjgowers @IAlibay I leave it to you to merge as I don't know if there's a particular reason to not merge it yet.

Was just waiting on CI, all good now :)

**Pull request merged**